src/main/kotlin/graphics/scenery/volumes/ColorMapPanel.kt

```diff
@@ -0,0 +1,332 @@
+            box.addItem(s)
+
+        if (target != null) {
+            box.selectedItem = selectAColorMapString
+            val currentColormap = JLabel("colormap: ")
+            this.add(currentColormap, "")
+            this.add(box, "grow")
+        }
+
+        box.addActionListener {
+            val item: String = box.selectedItem as String
+            if (target != null && item != selectAColorMapString) {
+                target.colormap = Colormap.get(item)
+                this.repaint()
+            }
+            colorMapEditor.loadColormap(Colormap.get(item))
+        }
+
+        val fc = JFileChooser()
+        fc.addChoosableFileFilter(PNGFileFilter())
+        fc.isAcceptAllFileFilterUsed = false
+
+        JButton("Load Color Map").also {
+            it.addActionListener {
+                val returnVal: Int = fc.showOpenDialog(this)
+                if (returnVal == JFileChooser.APPROVE_OPTION) loadFromFile(fc.selectedFile)
+            }
+            add(it)
+        }
+
+        JButton("Save Color Map").also {
+            it.addActionListener {
+                val option = fc.showSaveDialog(this)
+                if (option == JFileChooser.APPROVE_OPTION) {
+                    saveToFile (fc.selectedFile)
+                }
+            }
+            add(it,"wrap")
+        }
+    }
+
+    /**
+     * A filter to only select pngs.
+     */
+    class PNGFileFilter: FileFilter()
+    {
+        override fun accept(f: File): Boolean {
+            if (f.isDirectory()) {
+                return false
+            }
+            if (f.extension != "png") return false
+            return true
+        }
+
+        override fun getDescription(): String {
+            return "png"
+        }
+    }
+
+    /**
+     * Save a color map to file as a png.
+     */
+    fun saveToFile(file: File){
+        var file_ = file
+        if (FilenameUtils.getExtension(file.name).equals("png", ignoreCase = true)) {
+            // filename is OK as-is
+        } else {
+            file_ = File("$file.png") // append .png if "foo.jpg.png" is OK
+        }
+
+        try {
+            ImageIO.write(colorMapEditor.toImage(), "png", file_)
+        } catch (ioe: IOException) {
+            ioe.printStackTrace()
+        }
+    }
+
+    /**
+     * Load a color map from png.
+     */
+    fun loadFromFile(file: File){
+        var img: BufferedImage? = null
+        try {
+            img = ImageIO.read(file)
+        } catch (_: IllegalArgumentException){
+            logger.error("Could not find file ${file.path}")
+        } catch (e: IOException){
+            logger.error(e.toString())
+        }
+        if (img == null) return
+        val cm = Colormap.fromBuffer(Image.bufferedImageToRGBABuffer(img),img.width, img.height)
+        colorMapEditor.loadColormap(cm)
+    }
+}
+
+/**
+ * A Gui element to allow users to visually create or modify a color map
+ */
+class ColorMapEditor(var target:Volume? = null) : JPanel() {
```

Comment:
For separation of concerns, this should be in its own file.

Reply:
its now a sub class of color colorMapPanel.